tests/test_computeExPsf.py

```diff
@@ -0,0 +1,272 @@
+
+        tgp = GaussianProcessTreegp(
+            std=std,
+            correlation_length=correlation_length,
+            white_noise=white_noise,
+            mean=0.0,
+        )
+        tgp.fit(coord1, z1)
+        z = tgp.predict(coord)
+    else:
+        coord = coord1
+        z = z1
+
+    return pipeBase.Struct(coord=coord, z=z)
+
+
+class ComputeExPsfTestCase(lsst.utils.tests.TestCase):
+    """Test ComputeExPsfTask."""
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        output1 = generate_gaussian_random_field(
+            xmin=0,
+            xmax=2000,
+            ymin=0,
+            ymax=2000,
+            npoints=10000,
+            nmax=2000,
+            std=1.0,
+            correlation_length=200.0,
+            white_noise=0.01,
+            seed=42,
+            input_coord=None,
+        )
+
+        output2 = generate_gaussian_random_field(
+            xmin=0,
+            xmax=2000,
+            ymin=0,
+            ymax=2000,
+            npoints=10000,
+            nmax=2000,
+            std=1.0,
+            correlation_length=100.0,
+            white_noise=0.01,
+            seed=44,
+            input_coord=output1.coord,
+        )
+
+        self.coord1 = output1.coord
+        self.coord2 = output2.coord
+        self.de1 = output1.z
+        self.de2 = output2.z
+
+    def test_comp_ex_psf(self) -> None:
+        """Test that ex metric are compute and make sense."""
+
+        np.testing.assert_equal(self.coord1, self.coord2)
+
+        ra = self.coord1[:, 0]
+        dec = self.coord1[:, 1]
+
+        config = ComputeExPsfTask.ConfigClass()
+
+        config.min_sep = 0.01
+        config.max_sep = 5.0
+        config.nbins = 1
+        config.bin_type = "Linear"
+        config.sep_units = "arcmin"
+
+        task = ComputeExPsfTask(config)
+        output1 = task.run(self.de1, self.de2, ra, dec, units="arcmin")
+
+        # At small scale, expect the scalar two-point correlation function
+        # to be close to the input variance for de1 and de2. Cross correlation
+        # between de1 and de2 should be zeros are they are 2 indendant field.
+
+        np.testing.assert_allclose(output1.E1, 1.0, atol=2e-1)
+        np.testing.assert_allclose(output1.E2, 1.0, atol=2e-1)
+        np.testing.assert_allclose(output1.Ex, 0.0, atol=1e-1)
+
+        config = ComputeExPsfTask.ConfigClass()
+
+        config.min_sep = 600.0
+        config.max_sep = 1000.0
+        config.nbins = 1
+        config.bin_type = "Linear"
+        config.sep_units = "arcmin"
+
+        # At large scale, expect the scalar two-point correlation function to
+        # be all close to 0.
+
+        task = ComputeExPsfTask(config)
+        output2 = task.run(self.de1, self.de2, ra, dec, units="arcmin")
+
+        np.testing.assert_allclose(output2.E1, 0.0, atol=1e-1)
+        np.testing.assert_allclose(output2.E2, 0.0, atol=1e-1)
+        np.testing.assert_allclose(output2.Ex, 0.0, atol=1e-1)
```

Comment:
Could we include some intermediate scale as well? It'd be at least useful to check that the values from `E2` is about twice more than `E1` (or whatever factor we naively expect).